src/dmd/root/filename.d

```diff
@@ -761,10 +761,66 @@ nothrow:
         mem.xfree(cast(void*)str);
     }
 
-    extern (C++) const(char)* toChars() const pure
+    extern (C++) const(char)* toChars() const pure nothrow
     {
         return str;
     }
+
+    extern (D) const(char)[] toString() const pure nothrow
+    {
+        return str[0 .. strlen(str)];
```

Comment:
str can be null

Reply:
see https://github.com/dlang/dmd/pull/8550, which should be a separate PR anyway.